src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/OverflowDbNode.java

```diff
@@ -407,9 +401,26 @@ public int occurrenceToBlockOffset(Direction direction,
         + label + " edge to vertex " + adjacentVertex.id());
   }
 
+  /**
+   * Removes an 'edge', i.e. in reality it removes the information about the adjacent node from
+   * `adjacentVerticesWithProperties`. The corresponding elements will be set to `null`, i.e. we'll have holes.
+   * Note: this decrements the `offset` of the following edges in the same block by one, but that's ok because the only
+   * thing that matters is that the offset is identical for both connected nodes (assuming thread safety).
+   * @param blockOffset must have been initialized
+   */
+  protected void removeEdge(Direction direction, String label, VertexRef<OverflowDbNode> adjacentVertex, int blockOffset) {
+    if (blockOffset == OverflowDbEdge.UNINITIALIZED_BLOCK_OFFSET) {
+      throw new IllegalStateException("blockOffset not initialized for " + direction + " edge of type " + label +
+          " connected with node " + adjacentVertex.id + " of type " + adjacentVertex.label());
+    }
+    int offsetPos = getPositionInEdgeOffsets(direction, label);
+    int start = startIndex(offsetPos) + blockOffset;
+    int strideSize = getEdgeKeyCount(label) + 1;
 
-//  public void removeEdge(Direction direction, String label, VertexRef<OverflowDbNode> adjacentVertex, int blockOffset) {
-//  }
+    for (int i = 0; i < strideSize; i++) {
```

Comment:
Please do: for (int i = start; i < start + strideSize;...) {
  adja...[i] = null;
}

To stay consistent with the other for loops in this file.

Reply:
done